operator/src/main/java/oracle/kubernetes/operator/WaitForReadyStep.java

```diff
@@ -160,13 +168,19 @@ private void checkUpdatedResource(Packet packet, AsyncFiber fiber, Callback call
     fiber
         .createChildFiber()
         .start(
-            createReadAndIfReadyCheckStep(callback),
+            createReadAndIfReadyCheckStep(callback, packet),
             packet.clone(),
             null);
   }
 
-  private Step createReadAndIfReadyCheckStep(Callback callback) {
-    return createReadAsyncStep(getName(), getNamespace(), getDomainUid(), resumeIfReady(callback));
+  private Step createReadAndIfReadyCheckStep(Callback callback, Packet packet) {
```

Comment:
This doesn't follow our Step/Packet pattern. The Step is typically stateless or, at least, can be used for many different calls. The Packet is given on the Step.apply call. The logic in the else should happen in the Step.apply().

Reply:
I have made the suggested change, please let me know if I missed something. Thanks.